### PR TITLE
CreateTweet-14

### PR DESCRIPTION
**Pull request by bhball22**
Massiv Pull Request involving most htmls, Account models, Main Page urls, forms, and views

### CODE REVIEW

File: Main_Page/urls.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
-from django.urls import path
+from django.urls import path, re_path
```
Comment (bhball22) [inline Main_Page/urls.py:1]: the re_path can be removed, I actually never used it

File: Main_Page/views.py [hunk 2]
```diff
@@ -1,14 +1,33 @@
-from django.shortcuts import render
+import datetime
+from django.utils import timezone
+from django.shortcuts import render, get_object_or_404
```
Comment (bhball22) [inline Main_Page/views.py:3]: didn't use the get_object_or_404 so that can be removed

File: Account/models.py [hunk 3]
```diff
@@ -1,10 +1,33 @@
...
+
+    def visible_info(self):
+        if not self.tweet_text:
+            return
+        return self.tweet_text
+    
+class Generate_Tweet(ModelForm):
+    class Meta:
+        model = Tweet
+        fields =['tweet_text']
```
Comment (lmurdock12) [inline Account/models.py:24]: Does this Generate_Tweet class do anything? 

I see Generate_Tweet() is getting called in views.py but it looks like it is just calling the Generate_Tweet class in forms.py

File: Main_Page/templates/Main_Page/base.html [hunk 4]
```diff
@@ -63,6 +64,40 @@
...
+  font-size: 28px;
+  font-weight: bold;
+}
+
+.close:hover {
+  color: #aaa;
+  text-decoration: none;
+  cursor: pointer;
+}
+
```
Comment (lmurdock12) [inline Main_Page/templates/Main_Page/base.html:100]: Is any of this CSS used anywhere in this branch?

File: Main_Page/templates/Main_Page/index.html [hunk 5]
```diff
@@ -63,25 +68,34 @@ <h3 style="color: rgb(99, 99, 99);">@{{user.username}}</h3>
...
-  <h2>Tweet #2</h2>
-  <h5>Title description, Sep 2, 2017</h5>
-  <div class="fakeimg" style="height:60px;">Image</div>
-  <p>Some text..</p>
-  <p>Sunt in culpa qui officia deserunt mollit anim id est laborum consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco.</p>
-  <button class="like">Like</button>
-</div>
+  {% for post in explorescroll %}
+  {% if forloop.counter0 < 25 %}
+  {% load nbsp %}
```
Comment (lmurdock12) [inline Main_Page/templates/Main_Page/index.html:84]: Does the nbsp tag just add a single space in between tweets?

**Review by lmurdock12 — CHANGES_REQUESTED** [review]
Awesome work man. Looks nice and clean. Check out the couple comments I posted and let me know. Then we can merge in

File: Account/models.py (hunk 3 above)
Comment (bhball22) [reply Account/models.py:24]: Yes, this is the form, it is a way of having forms within Models

File: Main_Page/templates/Main_Page/base.html (hunk 4 above)
Comment (bhball22) [reply Main_Page/templates/Main_Page/base.html:100]: Not yet, this CSS is setup for the popup window for the login and register

File: Account/models.py (hunk 3 above)
Comment (bhball22) [reply Account/models.py:24]: The Generate_Tweet in forms calls this ModelForm in it's own way

Comment (lmurdock12) [reply Account/models.py:24]: Looks good.

Comment (bhball22) [reply Account/models.py:24]: Nevermind just tested without the ModelForm, it turns out we do not need it

**Review by lmurdock12 — APPROVED** [review]

**Pull request merged**